### PR TITLE
Resolves some low-hanging TODOs and tidy up TODOs

### PR DESCRIPTION
**Pull request by sebastiantoh**
https://github.com/hansel-app/hansel/issues/120

### CODE REVIEW

File: frontend/src/pages/LoginRegister/RegisterPage.vue [hunk 1]
```diff
@@ -58,21 +65,23 @@ export default defineComponent({
...
-        .catch((err) => console.log(err, "Failed to register"));
+      this.store
+        .dispatch("register", {
+          fullName: this.fullName,
+          username: this.username,
+          password: this.password,
+        })
+        .then(() => this.$router.push(HOME_ROUTE))
+        // TODO: maybe map some error messages to more user facing messages
+        // e.g HTTP response at 400 or 500 => Not connected to server or smth
```
Comment (ianyong) [inline frontend/src/pages/LoginRegister/RegisterPage.vue]: I guess we could, but then we would need to handle it at some API abstraction. Don't think we have the time for that. Let's just assume it should never happen (with our deployment setup)?

File: frontend/src/pages/LoginRegister/RegisterPage.vue [hunk 2]
```diff
@@ -27,7 +27,13 @@
         type="password"
         name="confirmPassword"
         placeholder="Confirm password"
-        :rules="[{ required: true, message: 'Confirm password is required' }]"
+        :rules="[
+          { required: true, message: 'Confirm password is required' },
+          {
+            validator: validatePasswordConfirmation,
```
Comment (ianyong) [inline frontend/src/pages/LoginRegister/RegisterPage.vue:33]: Side note: should we disable the button if any rules are not met?

Comment (sebastiantoh) [reply frontend/src/pages/LoginRegister/RegisterPage.vue:33]: hmm, not sure about this. Seems like most examples still allow you to click the submit button

**Review by ianyong — APPROVED** [review]
LGTM!

**Pull request merged**